daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala

```diff
@@ -471,6 +465,11 @@ class IdeClient(val compiledPackages: CompiledPackages) extends ScriptLedgerClie
             new RuntimeException("FATAL: Encountered scenario instruction getParty in DAML Script"))
       }
     }
+    // Reset the machine
```

Comment:
Should we do this in a finally block or something like that?

Reply:
The only `throw` is the `Unexpected node` exception, though to be safe it's probably better to use a finally block. At this point it's a bit weird though to use `Future.fromTry` in a `try` block. Maybe a follow-up refactoring would be to not use `fromTry` but just wrap the whole thing in `Future()`?

Reply:
Yeah wrapping everything in a future sounds reasonable.